hawkular-inventory-impl-tinkerpop-parent/hawkular-inventory-impl-tinkerpop-sql-provider/src/main/java/org/hawkular/inventory/impl/tinkerpop/sql/SqlGraphProvider.java

```diff
@@ -144,4 +146,16 @@ private static Object sampleValue(Class<?> type) {
             throw new IllegalArgumentException("Unhandled type of property: " + type);
         }
     }
+
+    @Override public RuntimeException translateException(RuntimeException inputException, CanonicalPath affectedPath) {
+        // Sqlg sends IllegalArgumentException when unique constraint is violated
+        // See SqlgElement.updateRow
+        if (inputException instanceof IllegalArgumentException
```

Comment:
This is coming from the backend (sqlg) right after we try the insert. A different exception is coming from Titan. We have a common logic but ask the providers for backend specific "questions" (like the "what does this exception mean?")..

This is of course brittle and would be best solved by making Sqlg throwing some more specific exception, but then again - Sqlg would have to do the same "magic" because it would have to translate the SQLExceptions thrown from the underlying database - those differ per db vendor, too.